clang/lib/Driver/Driver.cpp

```diff
@@ -1093,7 +1093,12 @@ void Driver::CreateOffloadingDeviceToolChains(Compilation &C,
   Arg *SYCLAddTargets =
       getArgRequiringSYCLRuntime(options::OPT_fsycl_add_targets_EQ);
   Arg *SYCLLink = getArgRequiringSYCLRuntime(options::OPT_fsycl_link_EQ);
-  Arg *SYCLfpga = getArgRequiringSYCLRuntime(options::OPT_fintelfpga);
+  Arg *SYCLfpga = C.getInputArgs().getLastArg(options::OPT_fintelfpga);
+
+  // Make -fintelfpga flag imply -fsycl.
+  if (SYCLfpga && !HasValidSYCLRuntime)
+    HasValidSYCLRuntime = true;
+
```

Comment:
huh, I just realized this - you can just add the `OPT_fintelfpga` check to the bool setting at line 1071.

Reply:
At L1080 - `HasValidSYCLRuntime` bool is also used to diagnose if SYCL offloading wasn't enabled for SYCL-specific options like `-fsycl-link-targets` , `-fsycl-add-targets`, `-fsycl-targets`,` -fsycl-link` and `-fsycl-host-compiler` and throw the error : 

error: 'sycl-specific-options-from-list-above' must be used in conjunction with '-fsycl' to enable offloading

Reply:
okay - that would imply moving the change up is the right one.  We only want to diagnose these situations when a SYCL enabling option isn't passed and is needed.